plugins/quota/rest.go

```diff
@@ -2,29 +2,125 @@ package quota
 	"github.com/gorilla/mux"
 )
 
-func commonValidator(w http.ResponseWriter, r *http.Request) {
-	ctx := r.Context()
-	volname := mux.Vars(r)["volname"]
-	v, err := volume.GetVolume(volname)
+//QuotadStart is the function to start the quota daemon
+func QuotadStart(c transaction.TxnCtx) error {
+	quotadDaemon, err := NewQuotad()
 	if err != nil {
-		restutils.SendHTTPError(ctx, w, http.StatusNotFound, errors.ErrVolNotFound.Error(), api.ErrCodeDefault)
-	} else if v.State != volume.VolStarted {
-		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, errors.ErrVolNotStarted.Error(), api.ErrCodeDefault)
+		return err
+	}
+	// Create pidfile dir if not exists
+	err = os.MkdirAll(path.Dir(quotadDaemon.pidfilepath), os.ModeDir|os.ModePerm)
+	if err != nil {
+		return err
 	}
+	// Create logFiledir dir
+	err = os.MkdirAll(path.Dir(quotadDaemon.logfilepath), os.ModeDir|os.ModePerm)
+	if err != nil {
+		return err
+	}
+	err = daemon.Start(quotadDaemon, true)
+	return nil
+}
+
+func storeVolume(c transaction.TxnCtx) error {
+
+	var volinfo volume.Volinfo
+	if err := c.Get("volinfo", &volinfo); err != nil {
+		return err
+	}
+
+	if err := volume.AddOrUpdateVolumeFunc(&volinfo); err != nil {
+		c.Logger().WithError(err).WithField(
+			"volume", volinfo.Name).Debug("storeVolume: failed to store volume info")
+		return err
+	}
+
+	if err := volgen.Generate(); err != nil {
+		c.Logger().WithError(err).WithField(
+			"volume", volinfo.Name).Debug("generateVolfiles: failed to generate volfiles")
+		return err
+	}
+
+	return nil
 }
 
 func quotaEnableHandler(w http.ResponseWriter, r *http.Request) {
-	// implement the help logic and send response back as below
+	// Collect inputs from URL
+	p := mux.Vars(r)
+	volName := p["volname"]
+
 	ctx := r.Context()
-	restutils.SendHTTPResponse(ctx, w, http.StatusOK, "todo: quota enable")
+	logger := gdctx.GetReqLogger(ctx)
+
+	// Validate volume existence
+	vol, err := volume.GetVolume(volName)
+	if err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusNotFound, errors.ErrVolNotFound.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	// Check if volume is started
+	if vol.State != volume.VolStarted {
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, errors.ErrVolNotStarted.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	txn := transaction.NewTxn(ctx)
+	defer txn.Cleanup()
+
+	if err := txn.Ctx.Set("volinfo", vol); err != nil {
+		logger.WithError(err).Error("failed to set volinfo in transaction context")
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	lock, unlock, err := transaction.CreateLockSteps(volName)
+	if err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	txn.Steps = []*transaction.Step{
+		lock,
+		{
+			DoFunc: "quota-enable.DaemonStart",
+			Nodes:  vol.Nodes(),
+		},
+		{
+			DoFunc: "quota-manage.UpdateVolinfo",
```

Comment:
You can reuse the existing, Refer https://github.com/gluster/glusterd2/blob/master/plugins/bitrot/rest.go#L155

Reply:
done.